src/db/mod.rs

```diff
@@ -211,7 +211,7 @@ impl Database {
     ) -> Result<Vec<String>> {
         let list_collections =
             ListCollections::new(self.name().to_string(), filter.into(), true, None);
-        let cursor = self
+        let cursor: Cursor<Document> = self
```

Comment:
I assume the reason this is needed is that `self` is a `Database<T>` here, and `T` isn't necessarily `Document`? I want to make sure that this isn't something that could happen with existing user code.

Reply:
The reason is because `Cursor::new` is now generic over `T`, so a `T` needs to be specified when it gets called. `Cursor::new` is `pub(crate)`, though, so it won't be an issue for existing code.

Reply:
Ah, right. Thanks for the clarification